core/lib/dal/src/lib.rs

```diff
@@ -47,7 +47,7 @@ pub mod fri_witness_generator_dal;
 pub mod healthcheck;
 mod instrument;
 mod metrics;
-mod models;
+pub mod models;
```

Comment:
This doesn't feel right. DAL models are majorly unsafe and leaking them publicly sounds like a bad idea.
If I understand correctly, proto models, on the other hand, are OK to be used publicly? 

If so, maybe it makes sense to move them out of `models` to their own top-level module (`proto_models`?) and make it public instead. If proto models are not meant to generally be public as well, we probably need to expose functionality related to encoding sizes somehow else.

cc @pompon0

Reply:
In this particular case the Payload struct (not the autogenerated proto code) is part of the public api of consensus_dal. The data of the payload (~miniblock) is stored in the db in a format unrelated to protobuf. The proto serialization is used in consensus, so it is part of the public API. In consensus_dal, we use protobuf encoding for hash validation in insert_certificate.

Reply:
Then probably it makes sense to create a module for actually publicly exposed API

Reply:
@popzxc @pompon0 I've extracted a top-level `consensus` module within `dal`. If later there would be additional public proto/Rust definitions for other domains, it can be rearranged.